### PR TITLE
Reimplement Rules

### PR DESCRIPTION
**Pull request by lsanchez79**

### CODE REVIEW

**Comment by privateducky** [comment]
just a PEP issue. good work otherwise!

**Comment by khyberspache** [comment]
well implemented!

File: app/utility/rule_set.py [hunk 1]
```diff
@@ -1,28 +1,29 @@
 import re
 from enum import Enum
 import ipaddress
-
-
+import logging
```
Comment (wbooth) [inline app/utility/rule_set.py]: delete

**Comment by wbooth** [comment]
@privateducky, the failed flake8 is from some other unrelated PRs that have been corrected. all flake8 issues on these sets of changes are fixed.

**Comment by wbooth** [comment]
@lsanchez79 please resolve the merge conflict here

File: app/objects/c_source.py [hunk 2]
```diff
@@ -15,6 +15,7 @@ def __init__(self, name, facts):
         super().__init__()
         self.name = name
```
Comment (wbooth) [inline app/objects/c_source.py]: add rules to init

Comment (wbooth) [reply app/objects/c_source.py]: def __init__(self, name, facts):

to

def __init__(self, name, facts, rules):

**Pull request merged**